File: C++/Program/Mahasiswa.cpp
/* 
    Saya Hestina Dwi Hartiwi NIM 2108077 mengerjakan soal Latihan 1 dalam Praktikum
    mata kuliah Desain dan Pemrograman Berorientasi Objek untuk keberkahan-Nya maka 
    saya tidak melakukan kecurangan seperti yang telah dispesifikasikan. Aamin.
*/

/* import library */
#include <string>
#include <iostream>

using namespace std;

/*--- Deklarasi Class ---*/
class Mahasiswa {
    /* Private Atributes */
    private:
        int code;
        string name, nim, major, faculty;
    public:
        /* Konstruktor */
        Mahasiswa()
        {
            this->code = '-';
            this->name = "";
            this->nim = "";
            this->major = "";
            this->faculty = "";
        }

        /* Getter and Setter */

        // Get Code
        int get_code()
        {
            return this->code;
        }

        // Get name
        string get_name()
        {
            return this->name;
        }

        // Get NIM
        string get_nim()
        {
            return this->nim;
        }

        // Get major
        string get_major()
        {
            return this->major;
        }

        // Get faculty
        string get_faculty()
        {
            return this->faculty;
        }
        
        // Set Code
        void set_code(int code)
        {
            this->code = code;
        }

        // Set name
        void set_name(string name)
        {
            this->name = name;
        }

        // Set NIM
        void set_nim(string nim)
        {
            this->nim = nim;
        }

        // Set Major
        void set_major(string major)
        {
            this->major = major;
        }

        // Set Faculty
        void set_faculty(string faculty)
        {
            this->faculty = faculty;
        }

        /* Destructor */
        ~Mahasiswa()
        {

        }
};

File: Java/Program/Mahasiswa.java
/* 
    Saya Hestina Dwi Hartiwi NIM 2108077 mengerjakan soal Latihan 1 dalam Praktikum
    mata kuliah Desain dan Pemrograman Berorientasi Objek untuk keberkahan-Nya maka 
    saya tidak melakukan kecurangan seperti yang telah dispesifikasikan. Aamin.
*/

/*--- Deklarasi Class ---*/
public class Mahasiswa 
{
    /* Private Atributes */
    private int code;       //---> untuk primary key
    private String name;    //---> untuk nama
    private String nim;     //---> untuk NIM
    private String major;   //---> untuk program studi
    private String faculty; //---> untuk fakultas

    /* Konstruktor */
    public Mahasiswa(int code, String name, String nim, String major, String faculty) {
        this.code = code;
        this.name = name;
        this.nim = nim;
        this.major = major;
        this.faculty = faculty;
    }

    /* Getter and Setter */
    
    //--- Get Code ---//
    public int getCode() {
        return this.code;
    }

    //--- Get Name ---//
    public String getName() {
        return this.name;
    }

    //--- Get NIM ---//
    public String getNim() {
        return this.nim;
    }

    //--- Get Major ---//
    public String getMajor() {
        return this.major;
    }

    //--- Get Faculty ---//
    public String getFaculty() {
        return this.faculty;
    }

    //--- Set Code ---//
    public void setCode(int code)
    {
        this.code = code;
    }

    //--- Set Name ---//
    public void setName(String name)
    {
        this.name = name;
    }

    //--- Set NIM ---//
    public void setNim(String nim)
    {
        this.nim = nim;
    }

    //--- Set Major ---//
    public void setMajor(String major)
    {
        this.major = major;
    }

    //--- Set Faculty ---//
    public void setFaculty(String faculty)
    {
        this.faculty = faculty;
    }
}